examples/1_advanced/12_advanced_transactions.rs

```diff
@@ -0,0 +1,91 @@
+// Copyright 2020-2025 IOTA Stiftung, Fondazione Links
+// SPDX-License-Identifier: Apache-2.0
+
+use examples::get_funded_client;
+use examples::get_memstorage;
+use examples::TEST_GAS_BUDGET;
+
+use anyhow::Context as _;
+use identity_iota::iota::rebased::client::IdentityClient;
+use identity_iota::iota::rebased::transaction_builder::MutGasDataRef;
+use identity_iota::iota::rebased::transaction_builder::Transaction as _;
+use identity_iota::iota::IotaDocument;
+use identity_iota::iota_interaction::IotaClientTrait as _;
+use identity_iota::iota_interaction::IotaKeySignature;
+use iota_sdk::rpc_types::IotaTransactionBlockResponseOptions;
+use iota_sdk::types::crypto::Signature;
+use iota_sdk::types::quorum_driver_types::ExecuteTransactionRequestType;
+use iota_sdk::types::transaction::GasData;
+use iota_sdk::types::transaction::Transaction;
+use iota_sdk::IotaClientBuilder;
+use iota_sdk::IOTA_COIN_TYPE;
+use secret_storage::Signer;
+
+/// This example demonstrates:
+/// 1. A user - Alice - can build a transaction that is sponsored by another user - Bob;
+/// 2. Deconstruct the transaction into its parts, to execute it manually through the SDK's IotaClient;
+/// 3. Apply the transaction's off-chain effects, from its on-chain ones.
+#[tokio::main]
+async fn main() -> anyhow::Result<()> {
+  let storage = get_memstorage()?;
```

Comment:
We don't have overlapping keys, but could we use separate storages to show the physical separation of the two?